snekmer/scripts/motif_motif.py

```diff
@@ -0,0 +1,156 @@
+
+config = snakemake.config
+
+# ---------------------------------------------------------
+# Run script
+# ---------------------------------------------------------
+
+# log script start time
+# start_time = datetime.now()
+
+# with open(snakemake.log[0], "a") as f:
+#     f.write(f"start time:\t{{start_time}}\n")
+
+# load input data
+with open(snakemake.input.matrix, "rb") as f:
+    data = pickle.load(f)
+
+# with open(snakemake.input.kmers, "rb") as f:
+#     kmers = f.readlines()
+
+with gzip.open(snakemake.input.weights, "rb") as f:
+    weights = pd.read_csv(f)
+
+with gzip.open(snakemake.input.scores, "rb") as f:
+    scores = pd.read_csv(f)
+    scores = scores.astype("float64")
+    scores = scores.to_numpy()
+
+
+with gzip.open(snakemake.input.kmers, "rb") as f:
+    kmers = pd.read_csv(f)
+
+# set category label name (e.g. "family")
+label = config["score"]["lname"] if str(config["score"]["lname"]) != "None" else "label"
+
+# with open(snakemake.input.model, "rb") as f:
+#     model = pickle.load(f)
+
+# with gzip.open(snakemake.input.vecs, "rb") as f:
+#     vecs=pd.read_csv(f)
+#     vecs.to_numpy
+
+# svm = LinearSVC(class_weight="balanced", random_state=None, max_iter=1000000)
+# vecs=np.array(data["sequence_vector"].astype(str).str.strip('[]').str.split(",").tolist(), dtype='float')
+# svm.fit(vecs, data[label])
+
+# prevent kmer NA being read as np.nan
+if config["k"] == 2:
+    kmers = kmers.fillna("NA")
+    scores = scores.fillna("NA")
+
+# kmers = weights['kmer'].values
+# scores = weights['sample'].values
+family = skm.utils.get_family(
+    skm.utils.split_file_ext(snakemake.input.weights)[0],
+    regex=config["input_file_regex"],
+)
+# scores = coeffs.iloc[score_index]
+scorer = skm.score.KmerScorer()
+
+gc.collect()
+
+# unit_score = max(scores)
+# print(unit_score)
+# for i in range(len(scores)):
+#     print(scores.iloc[i])
+#     scores.iloc[i] = scores.iloc[i]/unit_score
+
+# set number of permutations to test
+n_iter = config["motif"]["n"]
+
+
+# get kmers for this particular set of sequences
+# with open(snakemake.input.kmerobj, "rb") as f:
+#     kmerobj = pickle.load(f)
+
+
+# binary T/F for classification into family
+family = skm.utils.get_family(snakemake.wildcards.nb)
+binary_labels = [True if value == family else False for value in data[label]]
+
+# get alphabet name
+if config["alphabet"] in skm.alphabet.ALPHABET_ORDER.keys():
+    alphabet_name = skm.alphabet.ALPHABET_ORDER[config["alphabet"]].capitalize()
+else:
+    alphabet_name = str(config["alphabet"]).capitalize()
+
+
+# run permutations and score each
+
+score_matrix = kmers.rename(columns={"0": "kmer"})
+# score_matrix.rename(columns={"0": "kmer"}, inplace=True)
+score_array = pd.DataFrame.to_numpy(score_matrix)
+motif = skm.motif.SnekmerMotif()
+for file in snakemake.input.perm_scores:
+    with gzip.open(file) as f:
+        perm_scores = pd.DataFrame.to_numpy(pd.read_csv(f))
+    score_array = np.hstack((score_array, perm_scores))
+
+else:
```

Comment:
This actually pairs with the preceding for loop, although it looks like it isn't actually necessary with the current code. I think it's an artifact from before parallelizing the rescoring process, when that loop ran those iterations rather than just collecting the scores, so I'll see if I can remove it as part of cleaning up that file.

Reply:
Addressed in cdb04f78edc15d408f1c66a0b4efef3c721948cc.